lightning/src/blinded_path/payment.rs

```diff
@@ -92,6 +92,16 @@ enum BlindedPaymentTlvsRef<'a> {
 	Receive(&'a ReceiveTlvs),
 }
 
+/// A wrapper struct that stores the largest packet length in the given [`BlindedPath`].
+/// This helps us calculate the appropriate padding size for the [`BlindedPaymentTlvs`]
+/// at the time of writing them.
+pub(crate) struct PaymentLengthTlvs<'a> {
+	/// Length of the packet with the largest size in the [`BlindedPath`].
+	max_length: usize,
+	/// The current packet's TLVs.
+	tlvs: BlindedPaymentTlvsRef<'a>,
+}
+
```

Comment:
Instead of duplicating this and the `Writeable` implementation, let's make the wrapper struct generic by naming it simply `WithPadding` kinda like we do for `WithoutLength`.

Reply:
Sounds like a much cleaner way to handle the case!
Thanks for the suggestion!
Addressed in [pr3177.06](https://github.com/shaavan/rust-lightning/commits/pr3177.06)